datafusion/core/src/execution/runtime_env.rs

```diff
@@ -22,10 +22,7 @@ use crate::{
     error::Result,
     execution::disk_manager::{DiskManager, DiskManagerConfig},
 };
-use std::collections::HashMap;
 
-use crate::datasource::datasource::TableProviderFactory;
```

Comment:
The whole point of the PR is to remove the `datasource` dependency from the `runtime_env` module (as I am trying to move the `runtime_env` module into a new crate, `datafusion-execution`, that does not depend on `datasource`)

I plan to move object_store registry in a separate PR to keep the reviews smaller